common/djangoapps/enrollment/data.py

```diff
@@ -319,6 +319,7 @@ def get_roster(course_id):
             'name': UserProfile.objects.get(user=enrollment.user).name,
         }
         for enrollment in enrollments
+        if UserProfile.objects.get(user=enrollment.user)
```

Comment:
Why are we adding this? What case are you trying to avoid?

Won't this incur an additional database hit for each user?